tests/phpunit/tests/user.php

```diff
@@ -915,6 +915,78 @@ public function test_wp_insert_user_should_reject_user_login_that_matches_existi
 		$this->assertSame( 'existing_user_email_as_login', $user_id->get_error_code() );
 	}
 
+	/**
+	 * Tests that `wp_update_user()` allows updating a user when
+	 * the `user_login` and `user_email` are the same.
+	 *
+	 * @ticket 57967
+	 *
+	 * @covers ::wp_update_user
+	 */
+	public function test_wp_update_user_should_allow_user_login_with_same_user_email() {
+		$user_email = 'ababa@example.com';
+
+		$user_id = wp_insert_user(
+			array(
+				'user_login'    => $user_email,
+				'user_email'    => $user_email,
+				'user_pass'     => 'whatever',
+				'user_nicename' => 'whatever',
+			)
+		);
+
+		$existing_id = wp_update_user(
+			array(
+				'ID'         => $user_id,
+				'first_name' => 'Bob',
+			)
+		);
+
+		$this->assertNotWPError( $existing_id, 'A WP_Error object was not returned.' );
+		$this->assertSame( $existing_id, $user_id, 'The user ID to be updated and the existing user ID do not match.' );
+	}
+
+	/**
+	 * Tests that `wp_update_user()` rejects a `user_login` that matches an existing `user_email`.
+	 *
+	 * @ticket 57967
+	 *
+	 * @covers ::wp_update_user
+	 */
+	public function test_wp_update_user_should_reject_user_login_that_matches_existing_user_email() {
```

Comment:
This test passes _**before**_ applying the fix in this PR. This seems correct as it should reject for this dataset.